content/ember-data/v3/toJSON.md

```diff
@@ -0,0 +1,44 @@
+---
+id: record.toJSON
+title: Record toJSON usage
+until: '4.0.0'
+since: '3.12.0'
+---
+
+Users should avoid calling [`toJSON`](https://github.com/emberjs/data/blob/1be481a4924b2b4316c1cc151a58328c88903dcd/packages/store/addon/-private/system/model/model.js#L620) on a record instance since it uses the now deprecated `-default` serializer to create a JSON representation of a model.
+
+To clear this deprecation users may call record.serialize() or implement their own toJSON instead. The simplest 1:1 refactor is to import a serializer and define a `toJSON` method that returns the serialized data from the model, but users may want to consider implementing a custom "serialize" method that outputs relevant data.
+
+An example of the simple refactor is below:
+### before
+
+  `app/models/post.js`
+
+    import Model from '@ember-data/model';
+
+    export default Model.extend({});
+
+  `other app code`
+
+    const record = store.peekRecord('post');
+    // users the default serializer, will have a deprecation warning
+    const output = record.toJSON();
+
+### after
+  `app/models/post.js`
+
+    import Model from '@ember-data/model';
+    import { JSONAPISerializer } from '@ember-data/serializers';
+
+    export default Model.extend({
+      toJSON(options) {
+        let snapshot = this._internalModel.createSnapshot();
+        return JSONAPISerializer.serialize(snapshot, options);
```

Comment:
A few things:

1) This is attempting to call serialize as a static class method instead of as an instance method. `this.store.serializerFor` vs importing and calling is the way to go.

2) If we must show how to do this with private API then we would use `this._createSnapshot()` instead of via `_internalModel`. However, instead of recommending a private API we could recommend calling `this.serialize()` and munging the response as desired or iterating the record directly via `eachAttribute` `eachRelationship` etc.

Another alternative is to recommend using `recordDataFor` although I think this may not work yet (we should make sure this works nicely for folks in the near future though)